pkgs/by-name/pa/panda3ds/package.nix

```diff
@@ -0,0 +1,199 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeWrapper,
+
+  xorg,
+  kdePackages,
+  cmake,
+  pkg-config,
+  imagemagick,
+
+  # (vulkan is currently not supported with qtgui and will cause a segfault, so don't even bother trying to compile it)
+  withVulkan ? stdenv.hostPlatform.isLinux && !withQtGui,
+  withOpenGL ? stdenv.hostPlatform.isLinux || withQtGui,
+  withMetal ? stdenv.hostPlatform.isDarwin,
+  withLuaJIT ? true,
+  withQtGui ? false,
+  withQtWayland ? false, # (broken)
+  withDiscordRPC ? true,
+  withRenderdoc ? false,
+  sse4Support ? stdenv.hostPlatform.sse4_1Support,
+
+  SDL2,
+  libGL,
+  vulkan-headers,
+  vulkan-loader,
+  glslang,
+  wayland,
+  # misc. rt dep:
+  pulseaudio,
+
+  # msc. de-vendored deps
+  boost,
+  discord-rpc,
+  rapidjson, # required transitive: discord-rpc dependency
+  fdk_aac,
+  luajit,
+  libuv,
+
+# elfio,
+# imgui,
+# cryptopp,
+# xxHash,
+# httplib,
+# stb,
+# miniaudio,
+# toml11,
+# glm,
+# fmt,
+}:
+stdenv.mkDerivation {
+  pname = "panda3ds";
+  version = "0.8.0-unstable-2024-11-12";
+
+  src = fetchFromGitHub {
+    owner = "wheremyfoodat";
+    repo = "panda3ds";
+    rev = "b85dba277fb777b095623a24cd924d0725e1596b";
+    hash = "sha256-zKpz7c3frSzHbIKnTB88y9GJOqW31xQdgUDjoAdDTi8=";
+    fetchSubmodules = true;
+  };
+
+  patches = [
+    ./no-vendor.patch # de-vendor some dependencies
+    ./config-file-path.patch # move config file to the app data root
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
+    cmake
+    imagemagick
+  ] ++ (if withQtGui then [ kdePackages.wrapQtAppsHook ] else [ makeWrapper ]);
+
+  buildInputs =
+    [
+      SDL2
+      fdk_aac
+      boost
+    ]
+    ++ lib.optionals withQtGui (
+      [ kdePackages.qtbase ]
+      ++ (lib.optional (
+        lib.versionAtLeast kdePackages.qtbase.version "6" && stdenv.hostPlatform.isLinux && withQtWayland
+      ) kdePackages.qtwayland)
+    )
+    ++ lib.optionals withLuaJIT [
+      luajit
+      libuv
+    ]
+    ++ lib.optionals withDiscordRPC [
+      discord-rpc
+      rapidjson
+    ]
+    ++ lib.optionals withVulkan [
+      vulkan-headers
+      vulkan-loader
+      glslang
+    ]
+    ++ lib.optionals withOpenGL [ libGL ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      xorg.libX11
+      wayland
+    ];
+
+  # delete unused third_party deps
+  postUnpack = ''
+    rm -rf third_party/{SDL2,discord-rpc,fdk-aac,LuaJIT,boost}
+  '';
+
+  # fix missing ir/var/empty/constant_propagation_pass.cpp
+  postPatch = ''
+    mkdir -p third_party/dynarmic/src/dynarmic/ir/var
+    ln -s ../opt third_party/dynarmic/src/dynarmic/ir/var/empty
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    # main binary
+    install -Dm755 /build/source/build/Alber $out/bin/Alber
+
+    # desktop file
+    install -Dm444 $src/.github/Alber.desktop $out/share/applications/Alber.desktop
+
+    # icons
+    mkdir -p $out/share/icons/hicolor/256x256/apps
+    for size in 16 32 48 64 72 96 128 192 256 512 1024; do
```

Comment:
O_O Do we really need this many sizes for the icons? Usually just having 256x256 works fine, maybe 512x512